openapi/sled-agent.json

```diff
@@ -857,31 +857,22 @@
         ]
       },
       "IpNet": {
-        "description": "An `IpNet` represents an IP network, either IPv4 or IPv6.",
         "oneOf": [
           {
-            "type": "object",
-            "properties": {
-              "V4": {
+            "title": "v4",
+            "allOf": [
+              {
                 "$ref": "#/components/schemas/Ipv4Net"
               }
-            },
-            "required": [
-              "V4"
-            ],
-            "additionalProperties": false
+            ]
           },
           {
-            "type": "object",
-            "properties": {
-              "V6": {
+            "title": "v6",
+            "allOf": [
+              {
                 "$ref": "#/components/schemas/Ipv6Net"
               }
-            },
-            "required": [
-              "V6"
-            ],
-            "additionalProperties": false
+            ]
```

Comment:
What's the upshot of the `allOf` change?

Reply:
This is explained... some? .. in the code here:

https://github.com/oxidecomputer/omicron/pull/1024/files/9cdf6945c4e385a2ad62ca0319d2378bac6fbeea#diff-702b757e89b01c09790358016361ef6bd659081d6b4419810510f0b859366d11R1206-R1209

progenitor (via typify) will see this `title` field and use this as the label for an enum. So it will construct the type so that it matches the original `IpNet`.

I've opened an issue with schemars to have this in untagged schemas (which would do this automatically for us here): https://github.com/GREsau/schemars/issues/102